torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1421,6 +1426,22 @@ def sample_inputs_masked_select(op_info, device, dtype, requires_grad):
            supports_tensor_out=False,
            test_inplace_grad=False,
            sample_inputs_func=sample_inputs_broadcast_to),
+    UnaryUfuncInfo('bitwise_not',
+                   ref=np.bitwise_not,
+                   dtypes=integral_types_and(torch.bool),
+                   dtypesIfCPU=integral_types_and(torch.bool),
```

Comment:
Is specifying this for each device type actually necessary? I thought they inherited from dtype if they weren't specified?

Reply:
Since UnaryUfuncInfo defaults the type for these devices, we can either set them as the expected type or None

https://github.com/pytorch/pytorch/blob/9c0caf0384690cb67dcccb7066ece5184f72ca78/torch/testing/_internal/common_methods_invocations.py#L296-L318

Reply:
Oh that's right. Thank you for reminding me. Maybe we should change those defaults...

Reply:
Looking at that, I think many Ops are not being fully tested for all supported dtypes on ROCM

https://github.com/pytorch/pytorch/blob/9c0caf0384690cb67dcccb7066ece5184f72ca78/torch/testing/_internal/common_methods_invocations.py#L1302-L1306

https://github.com/pytorch/pytorch/blob/9c0caf0384690cb67dcccb7066ece5184f72ca78/torch/testing/_internal/common_methods_invocations.py#L1371-L1374

Reply:
Also I was wondering if,

If the following code in OpInfo
https://github.com/pytorch/pytorch/blob/9c0caf0384690cb67dcccb7066ece5184f72ca78/torch/testing/_internal/common_methods_invocations.py#L178-L180

Should be updated to because many times operators support slightly different dtypes between CPU and CUDA but they are mostly consistent between CUDA and ROCM,
```
self.dtypesIfCPU = set(dtypesIfCPU) if dtypesIfCPU is not None else self.dtypes 
self.dtypesIfCUDA = set(dtypesIfCUDA) if dtypesIfCUDA is not None else self.dtypes 
self.dtypesIfROCM = set(dtypesIfROCM) if dtypesIfROCM is not None else self.dtypesIfCUDA
```

Reply:
That's right. When I was developing the unary ufunc infos it wasn't easy to test against ROCm, so I was conservative about which dtypes were tested.

Reply:
Should we update it in this PR or in a seperate PR?

Also I have verified that we don't check for all types in ROCM build

https://ci.pytorch.org/jenkins/job/pytorch-builds/job/pytorch-linux-bionic-rocm3.10-py3.6-test1/7150/consoleFull

```
14:12:59   test_reference_numerics_asinh_cuda_float16 (__main__.TestUnaryUfuncsCUDA) ... ok (0.643s)
14:12:59   test_reference_numerics_asinh_cuda_float32 (__main__.TestUnaryUfuncsCUDA) ... ok (3.736s)
14:13:03   test_reference_numerics_asinh_cuda_float64 (__main__.TestUnaryUfuncsCUDA) ... ok (3.798s)
```

Reply:
I know. See here:

https://github.com/pytorch/pytorch/blob/594a66d778a660faed0b0fbbe1dd8c2c318707ff/test/test_ops.py#L28

This PR can update the ROCm dtype default as you've proposed as long as the CI passes. If it's a more involved change then I would separate it.